ci/templates/ci/reports.html

```diff
@@ -100,42 +120,64 @@ <h3 class="reports__heading">Report Tasks</h3>
 <script type="text/javascript">
 
     $(document).ready(function() {
-
-      if ($("#id_email_to").val() != '') {
-        var emails = $("#id_email_to").val().split(",");
-        $("#id_email_to").val('');
-        $.each(emails,function(i){
-          $("#id_email_to").after($(".email-to .email-to-group").clone().addClass("field-clone"));
-          $(".field-clone .email-fields").each(function(i){
-            $(this).attr('value', emails[i]);
+      // Replace encoded quotes with double quotes to make valid json
+      var forms = JSON.parse("{{ forms.keys }}".replace(/&#39;/g, '\"'))
+
+      $.each(forms, function(k){
+        // Get form and email_to field ids
+        var form_id = "#".concat(forms[k]);
+        var prefix = forms[k].replace("_form", "");
+        var email_to_id = "#".concat(prefix, "_email_to");
+
+        if ($(email_to_id).val() != '') {
+          var emails = $(email_to_id).val().split(",");
+          $(email_to_id).val('');
+          $.each(emails,function(i){
+            $(email_to_id).after($(".email-to .email-to-group").clone().addClass("field-clone"));
+            $(form_id).find(".field-clone .email-fields").each(function(i){
+              $(this).attr('value', emails[i]);
+            });
           });
-        });
         }
-      else {
-        $("#id_email_to").after($(".email-to .email-to-group").clone().addClass("field-clone"));
-      };
+        else {
+          $(email_to_id).after($(".email-to .email-to-group").clone().addClass("field-clone"));
+        };
 
-      $("#id_email_to").addClass("hide")
-      $(".field-clone:first .remove").replaceWith($(".email-to .add-more").show());
+        $(email_to_id).addClass("hide")
+        $(form_id).find(".field-clone:first .remove")
+            .replaceWith($(".email-to .add-more").clone().show());
+      });
 
       $(".add-more").click(function(){
-          $(".field-clone:last").after($(".email-to .email-to-group").clone());
+          $(this).parents(".control-group").after($(".email-to .email-to-group").clone());
```

Comment:
You can also declare $(this) in a variable var $this = $(this);  globally to the  document ready function I think instead on locally inside each function event.

Reply:
@Mitso I don't think I'm understanding properly. If I do 
```
$(document).ready(function() {
    var $this = $(this);
...
$(".add-more").click(function(){
    $this.parents(...
```
Then the `.add-more` button doesn't work any more? (I'm guessing because it can't find the element pointed to by `this`?)
How are you suggesting I do this?

Reply:
@KaitCrawford mhh ok cool, I think you can leave it then. Technicaly it's suppose to work because the it's declaration is shared globbaly and attached to local click event.